x/babylon/keeper/wasm.go

```diff
@@ -23,10 +23,7 @@ func (k Keeper) SendBeginBlockMsg(ctx sdk.Context) error {
 	headerInfo := ctx.HeaderInfo()
 	msg := contract.SudoMsg{
 		BeginBlockMsg: &contract.BeginBlock{
```

Comment:
There's no `BeginBlockMsg` anymore. Values are just flattened inside the `SudoMsg::BeginBlock`.

The same with`SudoMsg::EndBlock`.

Reply:
Tried, and looks like the marshaler already did flattening for us. Now I changed the sudo msg def to 

```go
// SudoMsg is a message sent from the Babylon module to a smart contract
type SudoMsg struct {
	BeginBlockMsg struct {
		HashHex    string `json:"hash_hex"`     // HashHex is the hash of the block in hex
		AppHashHex string `json:"app_hash_hex"` // AppHashHex is the app hash of the block in hex
	} `json:"begin_block,omitempty"`
	EndBlockMsg struct {
		HashHex    string `json:"hash_hex"`     // HashHex is the hash of the block in hex
		AppHashHex string `json:"app_hash_hex"` // AppHashHex is the app hash of the block in hex
	} `json:"end_block,omitempty"`
}
```

and construct the sudo msg accordingly. The e2e test gives the following 

```
--- FAIL: TestBabylonSDKTestSuite (1.75s)
    --- FAIL: TestBabylonSDKTestSuite/Test3BeginBlock (0.00s)
        main_test.go:134: 
                Error Trace:    /home/runchao/Projects/Babylon/babylon-sdk/tests/e2e/main_test.go:134
                Error:          Received unexpected error:
                                Error parsing into type babylon_apis::btc_staking_api::SudoMsg: Expected this character to start a JSON value.: execute wasm contract failed
                Test:           TestBabylonSDKTestSuite/Test3BeginBlock
    --- FAIL: TestBabylonSDKTestSuite/Test4EndBlock (0.00s)
        main_test.go:140: 
                Error Trace:    /home/runchao/Projects/Babylon/babylon-sdk/tests/e2e/main_test.go:140
                Error:          Received unexpected error:
                                Error parsing into type babylon_apis::btc_staking_api::SudoMsg: Expected this character to start a JSON value.: execute wasm contract failed
                Test:           TestBabylonSDKTestSuite/Test4EndBlock
```

So I suggest we keep the struct as is.

Reply:
I see, thanks for testing it.